models/evc.py

```diff
@@ -1029,6 +1030,16 @@ def get_id_from_cookie(cookie):
         evc_id = cookie - (settings.COOKIE_PREFIX << 56)
         return f"{evc_id:x}".zfill(14)
 
+    def set_flow_table_grou_id(self, flow_mod: dict, vlan) -> dict:
+        """Set table_group and table_id"""
+        table_group = "EPL" if vlan is None else "EVPL"
+        flow_mod["table_group"] = table_group
+        try:
+            flow_mod["table_id"] = self.table_group[table_group]
+        except KeyError:
+            flow_mod["table_id"] = self.table_group["base"]
```

Comment:
It's fine to have `"base"` table group as a valid group, but the way it's implemented here it can lead to surprises. If one day `"base"` is ever used in `mef_eline` then it'll likely try to derive something from the `flow_mod: dict` and probably ignore the vlan, since we don't have ever plans to use that one yet, I'll recommend that you remove the KeyError catch and don't assign to "base" yet, and one day if it's ever needed you can augment this func accordingly, OK?

Reply:
Good point @Alopalao. On `of_multi_table` we've specified as an array of enum string, so the caller might not specify all table groups, which is also a nicer API. So, with that said, it'd be safer that on `on_table_enabled` instead of assigning the entire payload, instead updating the keys that you're already validating, right?

Reply:
Got it.

Reply:
@Alopalao it's almost there. The current implementation still gives margin to still partly update a few groups, let's keep it either all or nothing to avoid surprises down the road. For instance, currently if `of_multi_table` were to accidentally send `{"evpl: 2", "invalid: 3", "epl: 4"}`, it would partly still updates "evpl" table group. Can you do it in another loop? To ensure this guarantee?

Reply:
Oh ye, changed it using `dict.update()`

Reply:
Nice. Even better.